mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCategoryConditionTest.java

```diff
@@ -0,0 +1,126 @@
+        Map<String, String> testProperties = super.getTestProperties();
+        testProperties.put("MCR.Metadata.Type.test", Boolean.TRUE.toString());
+        return testProperties;
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        MCRCategoryDAO instance = MCRCategoryDAOFactory.getInstance();
+        MCRCategoryImpl clazz = new MCRCategoryImpl();
+        clazz.setRootID("clazz");
+        clazz.setRootID("clazz");
+
+        clazz1 = new MCRCategoryImpl();
+        clazz1.setRootID("clazz");
+        clazz1.setCategID("clazz1");
+
+        clazz2 = new MCRCategoryImpl();
+        clazz2.setRootID("clazz");
+        clazz2.setCategID("clazz2");
+
+        instance.addCategory(null, clazz);
+        instance.addCategory(clazz.getId(), clazz1);
+        instance.addCategory(clazz.getId(), clazz2);
+
+    }
+
+    @Test
+    public void testConditionMatch() throws NoSuchFieldException, IllegalAccessException {
+        MCRSessionMgr.getCurrentSession().setUserInformation(MCRSystemUserInformation.getSuperUserInstance());
+
+        MCRObject object = new MCRObject();
+        MCRObjectID testId = MCRObjectID.getInstance("test_test_00000001");
+
+        MCRFactsHolder holder = new MCRFactsHolder(new ArrayList<>());
+        hackObjectIntoCache(object, testId);
+        holder.add(new MCRObjectIDFact("objid", testId.toString(), testId));
+
+        Collection<MCRCategoryID> collect = Stream.of(clazz1.getId()).collect(Collectors.toList());
+        MCRCategLinkServiceFactory.getInstance().setLinks(new MCRCategLinkReference(testId), collect);
+
+        MCRCategoryCondition categoryCondition = new MCRCategoryCondition();
+
+        categoryCondition.parse(new Element("classification").setText("clazz:clazz1"));
+        Assert.assertTrue("Object should be linked with clazz1",
+                categoryCondition.matches(holder));
+
+    }
+
+    @Test
+    public void testConditionNotMatch() throws NoSuchFieldException, IllegalAccessException {
+        MCRSessionMgr.getCurrentSession().setUserInformation(MCRSystemUserInformation.getSuperUserInstance());
+
+        MCRObject object = new MCRObject();
+        MCRObjectID testId = MCRObjectID.getInstance("test_test_00000001");
+
+        MCRFactsHolder holder = new MCRFactsHolder(new ArrayList<>());
+        hackObjectIntoCache(object, testId);
+        holder.add(new MCRObjectIDFact("objid", testId.toString(), testId));
+
+        Collection<MCRCategoryID> collect = Stream.of(clazz1.getId()).collect(Collectors.toList());
+        MCRCategLinkServiceFactory.getInstance().setLinks(new MCRCategLinkReference(testId), collect);
+
+        MCRCategoryCondition categoryCondition = new MCRCategoryCondition();
+
+        categoryCondition.parse(new Element("classification").setText("clazz:clazz2"));
+        Assert.assertFalse("Object not should be linked with clazz2",
+                categoryCondition.matches(holder));
+
+    }
+
+}
```

Comment:
Github already recognized "No linebreak" at the end
Please "auto-format" all new files